src/libraries/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationWriter.cs

```diff
@@ -12,10 +13,20 @@ namespace System.Xml.Serialization
 {
     internal sealed class ReflectionXmlSerializationWriter : XmlSerializationWriter
     {
-        private readonly XmlMapping _mapping;
+        internal static readonly TypeDesc StringTypeDesc;
+        internal static readonly TypeDesc QnameTypeDesc;
+
+        [RequiresUnreferencedCode("class GetTypeDesc")]
+        // Initialize in static constructor to suppress RequiresUnreferencedCode warning
+#pragma warning disable CA1810
+        static ReflectionXmlSerializationWriter()
```

Comment:
Can you fix this one too? Or maybe even better refactor it to use the one from the reader so it is only in 1 spot??